pallets/did/src/lib.rs

```diff
@@ -902,13 +954,24 @@ impl<T: Config> Pallet<T> {
 		operation: &DidAuthorizedCallOperationWithVerificationRelationship<T>,
 		signature: &DidSignature,
 	) -> Result<(), DidError> {
+		let current_block_number = <frame_system::Pallet<T>>::block_number();
+		// Before accessing the storage, we check if the provided block number is valid,
+		// i.e., if the current blockchain block is in the inclusive range
+		// [operation_block_number, operation_block_number + MaxBlocksTxValidity].
+		let allowed_range =
+			operation.block_number..=operation.block_number.saturating_add(T::MaxBlocksTxValidity::get());
+		ensure!(
+			allowed_range.contains(&current_block_number),
+			DidError::SignatureError(SignatureError::InvalidBlockNumber)
+		);
```

Comment:
I think a range is easier to understand, honestly, so I would rather keep it this way, especially since you don't seem to have a strong opinion about it either.

Reply:
To me, my proposal is easier to understand. Range definitely adds more complexity than just comparing two numbers.

Also, it just seems overkill to create a range for such a simple check and I would think it's a bit more resource heavy (but probably negligible small in Rust)?! 

Anyway, I do not have a strong opinion.

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/257/commits/f55b3d3c499288c02601439d6fddccb77e238104